fuse/plugins/truth_information/peak_truth.py

```diff
@@ -0,0 +1,160 @@
+            s2_photon_cut = photons_in_peaks[i]["photon_type"] == 2
+            ap_photon_cut = photons_in_peaks[i]["photon_type"] == 0
+
+            result['s1_photons_in_peak'][i] = np.sum(s1_photon_cut)
+            result['s2_photons_in_peak'][i] = np.sum(s2_photon_cut)
+            result['ap_photons_in_peak'][i] = np.sum(ap_photon_cut)
+
+            unique_contributing_clusters, photons_per_cluster = np.unique(photons_in_peaks[i]["cluster_id"], return_counts=True) 
+            unique_contributing_clusters_s1, photons_per_cluster_s1 = np.unique(photons_in_peaks[i][s1_photon_cut]["cluster_id"], return_counts=True) 
+            unique_contributing_clusters_s2, photons_per_cluster_s2 = np.unique(photons_in_peaks[i][s2_photon_cut]["cluster_id"], return_counts=True) 
+            unique_contributing_clusters_ap, photons_per_cluster_ap = np.unique(photons_in_peaks[i][ap_photon_cut]["cluster_id"], return_counts=True) 
+
+            result['number_of_contributing_clusters_s1'][i] = np.sum(unique_contributing_clusters_s1 > 0)
+            result['number_of_contributing_clusters_s2'][i] = np.sum(unique_contributing_clusters_s2 > 0)
+
+            contributing_clusters_s1 = get_cluster_information(interactions_in_roi, unique_contributing_clusters_s1)
+            contributing_clusters_s2 = get_cluster_information(interactions_in_roi, unique_contributing_clusters_s2)
+
+            if (result['s1_photons_in_peak'][i] + result['s2_photons_in_peak'][i]) > 0:
+                
+                positions_to_evaluate = ["x", "y", "z", "x_obs", "y_obs", "z_obs"]
+
+                for position in positions_to_evaluate:
+                    result_name = "average_" + position + "_of_contributing_clusters"
+
+                    result[result_name][i] = weighted_position_average(position,
+                        contributing_clusters_s1,
+                        contributing_clusters_s2,
+                        photons_per_cluster_s1,
+                        photons_per_cluster_s2,
+                        )
+            
+                #Assume that we calibrate or detector so that sum_s2_photons would give us the observed energy
+                energy_of_s2_photons_in_peak = photons_per_cluster_s2/contributing_clusters_s2["sum_s2_photons"] * contributing_clusters_s2["ed"]
+                #Same for S1 but with n_s1_photon_hits
+                energy_of_s1_photons_in_peak = photons_per_cluster_s1/contributing_clusters_s1["n_s1_photon_hits"] * contributing_clusters_s1["ed"]
+                #Sum up up the two: 
+                result['observable_energy_truth'][i] = np.sum(energy_of_s1_photons_in_peak) + np.sum(energy_of_s2_photons_in_peak)
+
+                result['raw_area_truth'][i] = np.sum(photons_in_peaks[i]["photon_gain"] / self.gains[photons_in_peaks[i]["channel"]])
+
+        return result
+
+        
+def get_cluster_information(interactions_in_roi, unique_contributing_clusters): 
```

Comment:
If `get_cluster_information` and `weighted_position_average` are functions only used in this `.py`, we might need to add a prefix `_` to it.